app/src/main/java/com/duckduckgo/app/notification/AppNotificationSender.kt

```diff
@@ -34,7 +34,7 @@ class AppNotificationSender(
 ) : NotificationSender {
 
     override suspend fun sendNotification(notification: SchedulableNotification) {
-        if (!notification.canShow()) {
+        if (!notification.canShow() || notificationDao.exists(notification.id)) {
```

Comment:
Does this mean if the Notification has been shown, we won't be able to show it again? Would that conflict with the `NetPWaitlistCodeNotificationWorker`?

Reply:
Yeah, the rational is that all `SchedulableNotification`(s) that currently override the `canShow` method do that `exists` check.
The reason is that none of the `SchedulableNotification` should show if the ID exists in that DAO. So in reality that check is an implementation detail of the `NotificationSender` that we're wrongly putting/repeating outside it.

That check should really be in the `sendNotification`, because otherwise the `SchedulableNotification`(s) need access to the DAO, which is in the :app module. Which would mean unnecessary extract the DAO to the `:browser-api` OR always place the `SchedulableNotification`s in the `:app` module. Neither good imo.

Reply:
Ahh got it 👍